cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/FindPluginHelper.java

```diff
@@ -70,12 +70,18 @@ public static Plugin findPlugin(ArtifactRepository artifactRepository,
       throw Throwables.propagate(e);
     }
 
+    CollectMacroEvaluator collectMacroEvaluator = new CollectMacroEvaluator();
+    MacroParser parser = new MacroParser(collectMacroEvaluator);
+
     // Just verify if all required properties are provided.
     // No type checking is done for now.
     for (PluginPropertyField field : pluginEntry.getValue().getProperties().values()) {
       Preconditions.checkArgument(!field.isRequired() || (properties.getProperties().containsKey(field.getName())),
                                   "Required property '%s' missing for plugin of type %s, name %s.",
                                   field.getName(), pluginType, pluginName);
+      if (field.isMacroSupported()) {
+        parser.parse(properties.getProperties().get(field.getName()));
```

Comment:
Was merged with this PR: https://github.com/caskdata/cdap/pull/6063/files
If null is passed to parse() then null will immediately be returned.